Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -178,13 +178,34 @@ void OpenALStream::SoundLoop()
   surround_capable = false;
 #endif
 
-  u32 ulFrequency = m_mixer->GetSampleRate();
-  numBuffers = SConfig::GetInstance().iLatency + 2;  // OpenAL requires a minimum of two buffers
+  u32 ul_frequency = m_mixer->GetSampleRate();
+  // OpenAL requires a minimum of two buffers
+  num_buffers = SConfig::GetInstance().iNumBuffers;
 
-  realtimeBuffer.reserve(OAL_MAX_SAMPLES * STEREO_CHANNELS);
-  sampleBuffer.reserve(OAL_MAX_SAMPLES * SURROUND_CHANNELS * OAL_MAX_BUFFERS);
-  uiBuffers.reserve(OAL_MAX_BUFFERS);
-  uiSource = 0;
+  if (num_buffers < 2)
+    num_buffers = 2;
+
+  // calculate latency (samples) per buffer
+  uint samples_per_buffer;
+  if (SConfig::GetInstance().iLatency > 0)
+  {
+	  samples_per_buffer = ul_frequency / 1000 * SConfig::GetInstance().iLatency / num_buffers;
+  }
+  else
+  {
+	  samples_per_buffer = ul_frequency / 1000 * 1 / num_buffers;
```

Comment:
I might be doing something wrong then, am I?
For me if ul_frequency is 48000 and iLatency is 1 then it will have a full latency of 48 samples, which is 1 ms. Actually, I guess I should limit iLatency's minimum to 3 ms, not 1 ms...